src/com/pentalog/pentastagiu/homework2/problema2/Basket.java

```diff
@@ -0,0 +1,41 @@
+package com.pentalog.pentastagiu.homework2.problema2;
+
+public class Basket {
+    private Customer customer;
+    private double totalPrice;
+
+    public Basket(Customer customer) {
+        this.customer = customer;
+    }
+
+    public void UpdateTotalPrice(double price)
+    {
+        totalPrice += price;
+    }
+
+    public double GetTotalPrice()
+    {
+        double percent = 0;
+        double total;
+        switch (customer.getMembership())
+        {
+            case GOLD:
+                percent = 0.2;
+                break;
+            case SILVER:
+                percent = 0.1;
+                break;
+            case NOMEMBERSHIP:
+                percent = 0.0;
+                break;
+        }
+        total = totalPrice - totalPrice*percent;
+        return total;
+    }
+
+    public void AddToBasket(Product product)
+    {
+        this.UpdateTotalPrice(product.getPrice());
```

Comment:
pentru cerintele problemei implementarea ta isi face treaba ok, dar mereu codul trebuie scris astfel incat sa ne fie usor sa il extindem; de exemplu ce se intampla daca in urmatoarea tema as cere sa faceti o evolutie in care sa puteti sa stergeti produse din cos pe baza id-ului lor, functionalitate f frecvent intalnita in siteurile de ecommerce; tu nu ai avea nicaieri o lista a acestor produse(array), ci doar costul lor total; ar fi fain sa implementezi si partea asta si sa incluzi calcularea pretului total pe baza listei de produse(astfel acoperi cazul in care un customer creeaza cosul acum, dar el da comanda peste 2 zile, timp in care pretul produselor s-a modificat)